platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -245,6 +257,51 @@ public Object getAttribute(String name) {
             return doPaosRequest(request, response);
         }
 
+        if (userAgentCheck && userAgentIsNotBrowser(httpRequest)) {
+            SecurityLogger.audit("Attempting to log client in as a legacy system.");
+            //if we get here, it is most likely an older DDF that is federating
+            //it isn't going to understand the redirect to the IdP and it doesn't support ECP
+            //so we need to fall back to other handlers to allow it to log in using PKI, Basic or Guest
```

Comment:
I would have expected any other handlers for the realm to get called anyways. I am wondering if there is a way to do this check that just picks up the configured authentication handlers for the target realm and gives them a chance to answer, rather than hardcoding PKI/Basic/Guest. Doesn't this also exclude falling back to anything that's not one of those three? Technically it is possible to deploy and configure additional authentication mechanisms.